ncm-dpmlfc/src/main/perl/DPMLFC/RuleBasedEditor.pm

```diff
@@ -0,0 +1,778 @@
+    return 1;
+  }
+  my $value = shift;
+  unless ( defined($value) ) {
+    $self->error("$function_name: 'value' argument missing");
+    return 1;
+  }
+  my $line_fmt = shift;
+  unless ( defined($line_fmt) ) {
+    $self->error("$function_name: 'line_fmt' argument missing");
+    return 1;
+  }
+
+  my $config_line = "";
+
+  if ( $line_fmt == LINE_FORMAT_PARAM ) {
+    $config_line = "$keyword=$value";
+  } elsif ( $line_fmt == LINE_FORMAT_ENVVAR ) {
+    $config_line = "export $keyword=$value";
+  } elsif ( $line_fmt == LINE_FORMAT_XRDCFG_SETENV ) {
+    $config_line = "setenv $keyword = $value";
+  } elsif ( $line_fmt == LINE_FORMAT_XRDCFG_SET ) {
+    $config_line = "set $keyword = $value";
+  } elsif ( $line_fmt == LINE_FORMAT_XRDCFG ) {
+    $config_line = $keyword;
+    $config_line .= " $value" if $value;
+    # In trust (shift.conf) format, there should be only one blank between
+    # tokens and no trailing spaces.
+    $config_line =~ s/\s\s+/ /g;
+    $config_line =~ s/\s+$//;
+  } else {
+    $self->error("$function_name: invalid line format ($line_fmt). Internal inconsistency.");
+  }
+
+  $self->debug(2,"$function_name: Configuration line : >>$config_line<<");
+  return $config_line;
+}
+
+
+# This function builds a pattern that will match an existing configuration line for
+# the configuration parameter specified. The pattern built takes into account the line format.
+# Every whitespace in the pattern (configuration parameter) are replaced by \s+.
+# If the line format is LINE_FORMAT_XRDCFG, no whitespace is
+# imposed at the end of the pattern, as these format can be used to write a configuration
+# directive as a keyword with no value.
+#
+# Arguments :
+#   config_param: parameter to update
+#   line_fmt: line format (see LINE_FORMAT_xxx constants)
+#   config_value: when defined, make it part of the pattern (used when multiple lines
+#                 with the same keyword are allowed)
+sub buildLinePattern () {
+  my $function_name = "buildLinePattern";
+  my $self = shift;
+
+  my $config_param = shift;
+  unless ( $config_param ) {
+    $self->error("$function_name: 'config_param' argument missing");
+    return undef;
+  }
+  my $line_fmt = shift;
+  unless ( defined($line_fmt) ) {
+    $self->error("$function_name: 'line_fmt' argument missing");
+    return undef;
+  }
+  my $config_value = shift;
+  if ( $config_value ) {
+    $self->debug(2,"$function_name: configuration value '$config_value' will be added to the pattern");
+    $config_value =~ s/\s+/\\s+/g;
+    $config_value =~ s/\-/\\-/g;
+    $config_value =~ s/\./\\./g;
+    $config_value =~ s/\*/\\*/g;
+    $config_value =~ s/\[/\\[/g;
+    $config_value =~ s/\]/\\]/g;
+    $config_value =~ s/\(/[(]/g;
+    $config_value =~ s/\)/\\)/g;
+    $config_value =~ s/\$/\\\$/g;
+    $config_value =~ s/\^/\\^/g;
+  } else {
+    $config_value = "";
+  }
+
+  $config_param =~ s/\s+/\\s+/g;
+  my $config_param_pattern;
+  if ( $line_fmt == LINE_FORMAT_PARAM ) {
```

Comment:
one advantage of using TT files also here is that at least one can easily verify that the patterns here actually match or are valid for the generated ones in method above.